src/xpcc/architecture/driver/atomic/queue.hpp

```diff
@@ -67,6 +67,9 @@ namespace xpcc
 			ALWAYS_INLINE bool
 			isFull() const;
 			
+			ALWAYS_INLINE bool
+			isNotFull() const { return not isFull(); }
```

Comment:
> I think Niklas introduced them and since he is the main developer of the code they start to prevail.

Yes, I introduced them intentionally to reduce confusion between logical and bitwise operators (`&&` vs. `&`, `||` vs. `|`, `!` vs `~`).

They are called Alternative Operators and are part of [the C++ language](http://en.cppreference.com/w/cpp/language/operator_alternative) (not part of C thought!).
We only use `and`, `or` and `not` though, and non of the other operators.
Yes, they are very rarely used even outside of embedded, and yes, most devs have never seen them 😛.

And prevail they should, as I think it significantly improves readability, especially for most logical "glue code" found in higher level drivers. I also find it tiring to visually parse the `!cond` for example, I find the `not ` in `not cond` to be a much larger visual "target" to find.
It also makes me think of code that uses it as "more abstract" or "less to the metal", because I associate it with Python.

I'm always amazed at some of the push back I get for this though 😞.
As if this is the most "outrageous" feature of xpcc, when it is _already_ so different from the rest of the frameworks out there.

Ultimately, as Thorsten said, this is a matter of "being used to it", which I think is the main reason why coding conventions exist. I don't think there is any science behind "this is better to read", at least I can't find any significant studies on this.

Also, as Thorsten said, since I'm one of the main developers, it's my way or the highway.
Now excuse me, I have to climb on my high horse. Bwahaha.